server/plugins/auth.js

```diff
@@ -61,92 +63,96 @@ export const register = (server, options, next) => {
-              }
+          if (!decoded.payload.sub.endsWith('@clients')) {
+            return { isValid: false };
+          }
 
-              return callback(null, true, decoded.payload);
-            });
-          });
-        } else if (decoded && decoded.payload && decoded.payload.iss === config('PUBLIC_WT_URL')) {
-          return jwt.verify(token, jwtOptions.dashboardAdmin.key, jwtOptions.dashboardAdmin.verifyOptions, (err) => {
-            if (err) {
-              return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-            }
+          const resourceServerKey = await getKeyAsync(decoded);
 
-            if (!decoded.payload.access_token || !decoded.payload.access_token.length) {
-              return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-            }
+          if (!resourceServerKey) {
+            return { isValid: false };
+          }
 
-            decoded.payload.scope = scopes.map(scope => scope.value); // eslint-disable-line no-param-reassign
-            return callback(null, true, decoded.payload);
-          });
+          // this can throw if there is an error
+          await jwtVerifyAsync(token, resourceServerKey, jwtOptions.resourceServer.verifyOptions);
+
+
+          if (decoded.payload.scope && typeof decoded.payload.scope === 'string') {
+            decoded.payload.scope = decoded.payload.scope.split(' '); // eslint-disable-line no-param-reassign
+          }
+
+          return { credentials: decoded.payload, isValid: true };
         }
-      }
 
-      return callback(null, false);
+        if (isDashboardAdminRequest) {
+          if (!decoded.payload.access_token || !decoded.payload.access_token.length) {
+            return { isValid: false };
+          }
+
+          // this can throw if there is an error
+          await jwtVerifyAsync(token, jwtOptions.dashboardAdmin.key, jwtOptions.dashboardAdmin.verifyOptions);
+
+          decoded.payload.scope = scopes.map(scope => scope.value); // eslint-disable-line no-param-reassign
+          return { credentials: decoded.payload, isValid: true };
+        }
+      } catch (error) {
+        return { isValid: false };
+      }
     }
   });
   server.auth.default('jwt');
   const session = {
-    register: tools.plugins.dashboardAdminSession,
+    plugin,
     options: {
       stateKey: 'authz-state',
       nonceKey: 'authz-nonce',
       sessionStorageKey: 'authz:apiToken',
       rta: config('AUTH0_RTA').replace('https://', ''),
       domain: config('AUTH0_DOMAIN'),
-      scopes: 'read:resource_servers create:resource_servers update:resource_servers delete:resource_servers read:clients read:connections read:rules create:rules update:rules update:rules_configs read:users',
+      // scopes: 'read:resource_servers create:resource_servers update:resource_servers delete:resource_servers read:clients read:connections read:rules create:rules update:rules update:rules_configs read:users',
```

Comment:
Remove comment?

Reply:
done in [46da9cc](https://github.com/auth0/auth0-authorization-extension/pull/387/commits/46da9cc5fb0dec70c6ea0e1fcb065ba66436c1e6)